src/uu/chmod/src/chmod.rs

```diff
@@ -37,13 +37,54 @@ mod options {
 
 #[uucore::main]
 pub fn uumain(args: impl uucore::Args) -> UResult<()> {
-    let mut args = args.collect_lossy();
+    let args = args.collect_lossy();
 
-    // Before we can parse 'args' with clap (and previously getopts),
-    // a possible MODE prefix '-' needs to be removed (e.g. "chmod -x FILE").
-    let mode_had_minus_prefix = mode::strip_minus_from_mode(&mut args);
+    // "-w -w" is the same as "-w,-w"
+    // "-w file -w" is the same as "-w,-w file"
+    // "-- -w file" is the same as "-w file"
+    let mut parsed_cmode_vec = Vec::new();
+    let double_hyphen = args.iter().find(|&x| x == "--");
```

Comment:
Maybe it's easier to make this a bool?

Reply:
I currently do it like that to get a `&String` of "--", as `take_while()` consumes it but clap needs it.

Maybe there is a better way?

Reply:
I could run a counter in the filter and then check if there is a double hyphen on that index. Would save a loop iteration

Reply:
I just meant something like this, basically what you currently have but with a bool instead of an option:
```rust
let double_hyphen_present = args.iter().any(|&x| x == "--");
// ...
if  double_hyphen_present {
    clean_args.push("--");
}
```

Reply:
Yep, I know. But clean_args is currently a Vec of String References. So I need an instance of "--" which I can reference. Otherwise, I might have to clone the Vec to get Strings instead of &Strings. That's basically the reason, but this might change when we don't collect the args anymore.

Reply:
Oh I see. Can we make the iterator over owned values instead? I.e. `args.into_iter()` instead of `args.iter()`?